File: apps/backend/src/modules/buildpack/buildpack.entity.ts
import {
  Collection,
  Entity,
  OneToMany,
  OneToOne,
  PrimaryKey,
  Property,
} from '@mikro-orm/core';
import { v4 } from 'uuid';
import { BuildpackVersion } from '../buildpack-version/buildpack-version.entity';
import { Deployment } from '../deployment/deployment.entity';

@Entity({ tableName: 'buildpacks' })
export class Buildpack {
  @PrimaryKey()
  id: string = v4();

  @Property()
  name: string;

  @OneToMany({
    entity: () => BuildpackVersion,
    mappedBy: 'buildpack',
    orphanRemoval: true,
  })
  versions: Collection<BuildpackVersion> = new Collection<BuildpackVersion>(
    this,
  );

  @OneToOne(() => Deployment)
  deployment: Deployment;

  @Property({ onCreate: () => new Date() })
  creationDate: Date = new Date();

  @Property({ onUpdate: () => new Date() })
  modificationDate: Date = new Date();
}

File: backend/src/main/java/com/esmo/empaas/entities/BuildpackEntity.java
package com.esmo.empaas.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.util.Date;
import java.util.List;

@Entity
@Table(name = "buildpacks")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class BuildpackEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(name = "name", unique = true)
    private String name;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true, mappedBy = "buildpack")
    private List<ApplicationEntity> applications;


    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true, mappedBy = "buildpack")
    private List<BuildpackVersionEntity> versions;

    @CreationTimestamp
    @Column(name = "creation_date")
    private Date creationDate;

    @UpdateTimestamp
    @Column(name = "modification_date")
    private Date modificationDate;
}
